nexus-tweaks.user.js

```diff
@@ -32,6 +32,391 @@ function timesCharExist(x, c){
+      person.maxHp = 39;
+      switch (parseInt(temp[1])) {
+        case 1:
+          person.currentHp = 39;
+          person.hpTooltip = 'Healthy';
+          break;
+        case 2:
+          person.currentHp = 29;
+          person.hpTooltip = 'Injured';
+          break;
+        case 3:
+          person.currentHp = 19;
+          person.hpTooltip = 'Serious';
+          break;
+        case 4:
+          person.currentHp = 9;
+          person.hpTooltip = 'Critical';
+          break;
+      }
+    }
+
+    // MP
+    temp = /title="(\d+)\/(\d+)\s+magic points"/.exec(
+        charHtml)
+    if (temp) {
+      person.mpVisible = true;
+      person.currentMp = parseInt(temp[1]);
+      person.maxMp = parseInt(temp[2]);
+      person.hpTooltip += ', ' + person.currentMp + '/' + person.maxMp + ' MP';
+    } else {
+      person.currentMp = 0;
+      person.maxMp = 0;
+    }
+
+    // Morality
+    temp = /title="(\w+)\s+alignment"/.exec(charHtml)
+    if (temp) {
+      person.morality = temp[1];
+    }
+
+    // Sorcerers Might
+    temp = /SM (\d+) min/.exec(charHtml);
+    if (temp) {
+      person.smDuration = temp[1];
+      var smStatus = {
+        'abbrev': 'SM ' + person.smDuration,
+        'title': 'Sorcerer\'s Might (' + person.smDuration + ' minutes)',
+        'class': 'status-tag-negative'
+      };
+      person.statusTags.push(smStatus);
+    }
+
+    // Get all externally visible statuses (displayed as images in vanilla)
+    // This will fetch hidden, poison, agony curse etc. (see below)
+    var statusRegExp = /<img.+?title="(.+?)".+?src="(.+?)"/;
+    var statusMatch = statusRegExp.exec(charHtml);
+
+    while (statusMatch != null) {
+      // ignore health, magic and SM, since already handled
+      if (statusMatch[2].indexOf("HealthBar") < 0
+          && statusMatch[2].indexOf("MagicBar") < 0
+          && statusMatch[1].indexOf("Sorcerers Might") < 0) {
+        var newStatus = {
+          'abbrev': statusMatch[1].substring(0, 2),
+          'title': statusMatch[1],
+          'class': 'status-tag-standard'
+        };
+        checkStatus(newStatus, 'Evil', 'E', 'status-tag-ev');
+        checkStatus(newStatus, 'Neutral', 'N', 'status-tag-nt');
+        checkStatus(newStatus, 'Good', 'G', 'status-tag-gd');
+        checkStatus(newStatus, 'Hidden', 'H', 'status-tag-positive');
+        checkStatus(newStatus, 'Flying', 'F', 'status-tag-positive');
+        checkStatus(newStatus, 'Hellfire', 'HF', 'status-tag-negative');
+        checkStatus(newStatus, 'Defiler Poison', 'DP', 'status-tag-negative');
+        checkStatus(newStatus, 'cursed by', 'AC', 'status-tag-negative');
+        person.statusTags.push(newStatus);
+      }
+      // remove processed status, move on to next one
+      charHtml = charHtml.replace(statusMatch[0], '');
+      statusMatch = statusRegExp.exec(charHtml);
+    }
+    return person;
+  }
+
+  const characterListTweaks = async (mod) => {
```

Comment:
nit: Method name should be more indicative of which it is, e.g. "visualCharacterList".
There will be a different method for the old libC style character list too.